hzerrors/codes.go

```diff
@@ -16,100 +16,107 @@
-	ErrorCodeOperationTimeout                 ErrorCode = 40
-	ErrorCodePartitionMigrating               ErrorCode = 41
-	ErrorCodeQuery                            ErrorCode = 42
-	ErrorCodeQueryResultSizeExceeded          ErrorCode = 43
-	ErrorCodeQuorum                           ErrorCode = 44
-	ErrorCodeReachedMaxSize                   ErrorCode = 45
-	ErrorCodeRejectedExecution                ErrorCode = 46
-	ErrorCodeRemoteMapReduce                  ErrorCode = 47
-	ErrorCodeResponseAlreadySent              ErrorCode = 48
-	ErrorCodeRetryableHazelcast               ErrorCode = 49
-	ErrorCodeRetryableIO                      ErrorCode = 50
-	ErrorCodeRuntime                          ErrorCode = 51
-	ErrorCodeSecurity                         ErrorCode = 52
-	ErrorCodeSocket                           ErrorCode = 53
-	ErrorCodeStaleSequence                    ErrorCode = 54
-	ErrorCodeTargetDisconnected               ErrorCode = 55
-	ErrorCodeTargetNotMember                  ErrorCode = 56
-	ErrorCodeTimeout                          ErrorCode = 57
-	ErrorCodeTopicOverload                    ErrorCode = 58
-	ErrorCodeTopologyChanged                  ErrorCode = 59
-	ErrorCodeTransaction                      ErrorCode = 60
-	ErrorCodeTransactionNotActive             ErrorCode = 61
-	ErrorCodeTransactionTimedOut              ErrorCode = 62
-	ErrorCodeURISyntax                        ErrorCode = 63
-	ErrorCodeUTFDataFormat                    ErrorCode = 64
-	ErrorCodeUnsupportedOperation             ErrorCode = 65
-	ErrorCodeWrongTarget                      ErrorCode = 66
-	ErrorCodeXA                               ErrorCode = 67
-	ErrorCodeAccessControl                    ErrorCode = 68
-	ErrorCodeLogin                            ErrorCode = 69
-	ErrorCodeUnsupportedCallback              ErrorCode = 70
-	ErrorCodeNoDataMember                     ErrorCode = 71
-	ErrorCodeReplicatedMapCantBeCreated       ErrorCode = 72
-	ErrorCodeMaxMessageSizeExceeded           ErrorCode = 73
-	ErrorCodeWANReplicationQueueFull          ErrorCode = 74
-	ErrorCodeAssertionError                   ErrorCode = 75
-	ErrorCodeOutOfMemoryError                 ErrorCode = 76
-	ErrorCodeStackOverflowError               ErrorCode = 77
-	ErrorCodeNativeOutOfMemoryError           ErrorCode = 78
-	ErrorCodeNotFound                         ErrorCode = 79
-	ErrorCodeStaleTaskID                      ErrorCode = 80
-	ErrorCodeDuplicateTask                    ErrorCode = 81
-	ErrorCodeStaleTask                        ErrorCode = 82
-	ErrorCodeLocalMemberReset                 ErrorCode = 83
-	ErrorCodeIndeterminateOperationState      ErrorCode = 84
-	ErrorCodeFlakeIDNodeIDOutOfRangeException ErrorCode = 85
-	ErrorCodeTargetNotReplicaException        ErrorCode = 86
-	ErrorCodeMutationDisallowedException      ErrorCode = 87
-	ErrorCodeConsistencyLostException         ErrorCode = 88
+	errorCodeUndefined                        errorCode = 0
```

Comment:
Are these error codes the same as the ones used int https://github.com/hazelcast/hazelcast-go-client/blob/master/hzerrors/server_error.go#L20 ?

Reply:
The `ServerError`s error code is coming from the decoder, then it is matched at https://github.com/hazelcast/hazelcast-go-client/blob/a68b383a9367e604696bfe736f4bba8ebf504f2d/hzerrors/error.go#L291

Reply:
But this is also a public API, right? I mean the this method https://github.com/hazelcast/hazelcast-go-client/blob/master/hzerrors/server_error.go#L45 

If we are making this public, I think we should also make the error codes public. In the previous PR, I just wanted to say that this information is private in the Java client

Reply:
`ErrorCode` is public, since it has to be used in `socketReadLoop` in `internal/cluster`. The error stuff should be refactored to use plain Go errors, instead of structs, but I'll do that in a separate PR.